src/librustc/hir/lowering.rs

```diff
@@ -1922,7 +1922,7 @@ impl<'a> LoweringContext<'a> {
             variadic: decl.variadic,
             has_implicit_self: decl.inputs.get(0).map_or(false, |arg| match arg.ty.node {
                 TyKind::ImplicitSelf => true,
-                TyKind::Rptr(_, ref mt) => mt.ty.node == TyKind::ImplicitSelf,
+                TyKind::Rptr(_, ref mt) => mt.ty.node.is_implicit_self(),
```

Comment:
I think this should just use an inline `match` - since it's really the `TyKind::ImplicitSelf => true,` line above, applied to the pointee of the reference.